### PR TITLE
Gutenboarding: Remove all unused bits and reorganize the rest

### PR DESCRIPTION
**Pull request by tyxla**
While working on #74399 I noticed that none of Gutenboarding is accessible, since it would always redirect to `/start`:

https://github.com/Automattic/wp-calypso/blob/3626bab4dd8d7785adc0b13fa501499220be1774/client/server/pages/index.js#L901-L915

Regardless that we won't be using it, we are building the Gutenboarding entry point along with all of its code.

Might be easier to review on a per-commit basis, since all changes that aren't removals are in separate commits.

## Proposed Changes

This PR removes the unused Gutenboarding entry point, along with all unused code from there.

The little remaining code is being moved where it is currently used and makes more sense.

## Testing Instructions

* Verify that you can't reach any Gutenboarding route (`/new*`)
* Smoke test new user signup and existing user site creation flow. 
* Verify all checks are green.

### CODE REVIEW

File: client/landing/gutenboarding/lib/analytics/index.ts
Comment (noahtallen) [inline client/landing/gutenboarding/lib/analytics/index.ts:1]: Why do we still need this file? (I might bee seeing it out of order in the commit review!)

File: client/landing/stepper/stores.ts [hunk 1]
```diff
@@ -24,3 +25,5 @@ export const USER_STORE = User.register( {
 } );
 
 export const AUTOMATED_ELIGIBILITY_STORE = AutomatedTransferEligibility.register();
+
+export const PLANS_STORE = Plans.register();
```
Comment (noahtallen) [inline client/landing/stepper/stores.ts:29]: Good example of how importing directly from the actual store is useful! This change probably wouldn't have been needed if the code was already using the new redux pattern :)

**Review by noahtallen — APPROVED** [review]
No concerns on my end!

That said, I'd love to get confirmation from the product side that gutenboarding is fully dead and that there aren't any future plans for it.

**Review by worldomonation — APPROVED** [review]
I know I'm not part of the reviewers list, but wanted to say thank you for going back and removing deprecated code. It's an underappreciated task. 

I do not believe any E2E tests remain that uses Gutenboarding, but POMs do remain. I will create another PR based on this branch that cleans those POMs up.

**Review by sirbrillig — APPROVED** [review]
I don't have any info on the usage of this flow, but from a Payments perspective, removing it is great! We've had to do a bunch of extra work to support Gutenboarding in addition to signup in past years so this would mean one fewer thing to maintain.

File: client/blocks/login/index.jsx [hunk 2]
```diff
@@ -45,6 +44,27 @@ import ErrorNotice from './error-notice';
...
+	if ( queryParamIndex === -1 ) {
+		return false;
+	}
+	const searchParams = new URLSearchParams(
+		decodeURIComponent( urlString.slice( queryParamIndex ) )
+	);
+	const anchorFmPodcastId = searchParams.get( 'anchor_podcast' );
+	return Boolean( anchorFmPodcastId && anchorFmPodcastId.match( /^[0-9a-f]{7,8}$/i ) );
+}
+
```
Comment (alshakero) [inline client/blocks/login/index.jsx:67]: We have another one of these [here](https://github.com/Automattic/wp-calypso/blob/remove/gutenboarding/client/landing/stepper/declarative-flow/anchor-fm-flow.ts#L14). Maybe worth putting in the `onboarding` package.

**Review by alshakero — APPROVED** [review]
Did a quick sweep, and it's much easier to review than I thought! LGTM!

Disclaimer: I didn't test. Only reviewed.

**Comment by alshakero** [comment]
Not gonna lie, having spent months working on this, it breaks my heart a little.

**Comment by tyxla** [comment]
Thank you all for taking a look 🙌 

> Not gonna lie, having spent months working on this, it breaks my heart a little.

I totally sympathize. It served us really well!

File: client/blocks/login/index.jsx (hunk 2 above)
Comment (tyxla) [reply client/blocks/login/index.jsx:67]: Given that those are different functions, I preferred not to introduce additional dependencies to the Login block used in multiple places, but this is definitely something worth arranging at some point.

**Pull request merged**